File: src/main/java/me/syari/ss/discord/internal/handle/GuildCreateHandler.java
package me.syari.ss.discord.internal.handle;

import me.syari.ss.discord.api.utils.data.DataObject;
import me.syari.ss.discord.internal.JDA;
import me.syari.ss.discord.internal.entities.Guild;
import org.jetbrains.annotations.NotNull;

public class GuildCreateHandler extends SocketHandler {
    public GuildCreateHandler(JDA api) {
        super(api);
    }

    @Override
    protected Long handleInternally(@NotNull DataObject content) {
        System.out.println(">> GuildCreateHandler");
        final long id = content.getLong("id");
        Guild guild = getJDA().getGuildById(id);
        if (guild == null) {
            getJDA().getGuildSetupController().onCreate(id, content);
        }

        return null;
    }
}

File: src/main/kotlin/me/syari/ss/discord/internal/handle/GuildCreateHandler.kt
package me.syari.ss.discord.internal.handle

import me.syari.ss.discord.api.data.DataContainer
import me.syari.ss.discord.internal.entities.Guild.Companion.contains

object GuildCreateHandler: SocketHandler() {
    override fun handleInternally(content: DataContainer): Long? {
        println(">> GuildCreateHandler")
        val id = content.getLongOrThrow("id")
        if (!contains(id)) GuildSetupController.onCreate(id, content)
        return null
    }
}